### ISSUE TITLE
fortios_firewall_policy - lists in module

### ISSUE THREAD

**Issue by adgerth** [fortinet-ansible-dev/ansible-galaxy-fortios-collection#102]
Hi all,

I'm trying to get a list of items working with the parameters 'dstintf' and 'dstaddr'.
So far, I've got this playbook:
```yaml
---
- hosts: firewall
  gather_facts: false
  collections:
    - fortinet.fortios
  connection: httpapi
  vars:
    ansible_httpapi_use_ssl: yes
    ansible_httpapi_validate_certs: no
    ansible_httpapi_port: 10443
  tasks:
  - name: create and delete IPv4 firewall policies
    fortinet.fortios.fortios_firewall_policy:
      vdom: TEST
      state: present
      access_token: "{{ REDACTED }}"
      firewall_policy:
        name: allow ssh, http and https from VLAN10 to VLAN20 and VLAN30
        srcintf:
          - name: VLAN10
        dstintf:
          - name: VLAN20
          - name: VLAN30
        srcaddr:
          - name: VLAN10 - Subnet
        dstaddr:
          - name: VLAN20 - Subnet
          - name: VLAN30 - Subnet
        action: accept
        policyid: 5
        nat: disable
        logtraffic: all
        schedule: always
        service:
          - name: SSH
          - name: HTTP
          - name: HTTPS
        ssl_ssh_profile: certificate-inspection
...
```
Expected bahvior:
Works

Actual behavior:
```json
"meta": {"build": 1190, "error": -5, "http_method": "POST", "http_status": 500, "name": "policy", "path": "firewall", "revision": "2037bce421d03febfa1f2a6c12f78eb1", "revision_changed": false, "serial": "REDACTED", "status": "error", "vdom": "TEST", "version": "v6.2.7"}, "msg": "Error in repo", "version_check_warning": {"matched": false, "mismatches": ["module fortios_firewall_policy not supported until in v6.0.0"], "system_version": "undefined"}}
```

When using one single Item for the parameters(dstintf, dstaddr), the policy works.
In the documentation, it says the parameters are of type: list
The list of services works regardless so I suspect they are specified correctly.

Regarding the version mismatch, I can actually get the version using the fortios_configuration_fact module in a playbook like this:
```yaml
- name: getting configuration facts from fortigates
  fortinet.fortios.fortios_configuration_fact:
    vdom: "TEST"
    access_token: "{{ REDACTED }}"
    selector: firewall_address
  register: fortinet_facts
 
- name: setting system_version fact
  set_fact:
    system_version: "{{ fortinet_facts['meta']['version'] }}"
 
- name: print version fact
  debug:
    var: fortinet_facts['meta']['version']
```
Output:
```
TASK [getting configuration facts from fortigates] ************************************************
ok: [fortinet.example.tld]

TASK [setting system_version fact] ****************************************************************
ok: [fortinet.example.tld]

TASK [print version fact] *************************************************************************
ok: [fortinet.example.tld] => {
    "fortinet_facts['meta']['version']": "v6.2.7"
}
```

**Comment by JieX19** [comment]
Hi @adgerth,
I guess the error is caused by lacking of available dstintf or dstaddr. I tried the same playbook and everything works fine. Please check the interface and firewall_address first and make sure the values in dstintf and dstaddr are correct. 

![Inkedresult_LI](https://user-images.githubusercontent.com/59745792/117218897-f1f6c080-adb8-11eb-9d6b-082114e76d7e.jpg)

**Comment by adgerth** [comment]
Hi @JieX19 
thanks for your answer.  
I'm fairly certain that they're present and correct since I can create rules with each of them when using one item in the list for dstintf, srcintf, dstaddr, and srcaddr. Also wenn creating and modifying them using ansble, they return 'OK' for all interfaces I've defined.

**Comment by JieX19** [comment]
Hi @adgerth,

I cannot reproduce the error. One more thing you need to verify is that if there's a duplicate name on your device, I mean **name: allow ssh, http and https from VLAN10 to VLAN20 and VLAN30**, this would caused an error. 

My runnable playbook:
```          
- hosts: fortigate
  connection: httpapi
  collections:
    - fortinet.fortios
  vars:
    ansible_httpapi_use_ssl: yes
    ansible_httpapi_validate_certs: no
    ansible_httpapi_port: 443
    vdom: "root"
  tasks:
  - name: create IPv4 firewall policy
    fortios_firewall_policy:
      vdom: "{{ vdom }}"
      state: "present"
      access_token: "{{ fortios_access_token }}"
      firewall_policy:
        name: VLAN_10 to VLAN_20 and VLAN_30
        srcintf:
          - name: VLAN_10
        dstintf:
          - name: VLAN_20
          - name: VLAN_30
        srcaddr:
          - name: VLAN_10 address
        dstaddr:
          - name: VLAN_20 address
          - name: VLAN_30 address
        action: accept
        policyid: 8
        nat: disable
        logtraffic: all
        schedule: always
        service:
          - name: HTTP
          - name: HTTPS
          - name: SSH
        ssl_ssh_profile: certificate-inspection
```

If possible, can you please provide me the debugging log?
Verbose output can help with debugging. To enable verbose debugging, use the following commands in the FortiGate CLI:

diagnose debug enable
diagnose debug application httpsd -1

**Comment by adgerth** [comment]
Hi @JieX19
I've set the name to 'allow some traffic' but still the same result.
I've attached the ansible output (job.log) and the fw debug logs (debug.log)
[debug.log](https://github.com/fortinet-ansible-dev/ansible-galaxy-fortios-collection/files/6450575/debug.log)
[job.log](https://github.com/fortinet-ansible-dev/ansible-galaxy-fortios-collection/files/6450576/job.log)

**Comment by adgerth** [comment]
Hi @JieX19 

I'm already using version 2.0.1, here's my collections/requirements.yml:
```
$ cat collections/requirements.yml
---
collections:
  - name: fortinet.fortios
    version: 2.0.1
    source: https://galaxy.ansible.com
...
```
Here's the additional info you requested:
```
$ ansible --version
ansible 2.9.19
  config file = /home/example.tld/adgerth/ansible/fortinet/ansible.cfg
  configured module search path = ['/home/example.tld/adgerth/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Aug 18 2020, 08:33:21) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]
```
I don't know if it helps, but we're using a custom ansible.cfg with the playbooks:
```
$ cat ansible.cfg
[defaults]
inventory = inventory
retry_files_enabled = False
roles_path = roles
collections_paths = collections
```

**Comment by JieX19** [comment]
The environment seems good. How is the results after running the playbook in the new environment? Does it still override the previous values and just take the last one in the end?

**Comment by adgerth** [comment]
The error remains the same since I've already been using this setup the whole time working on this.
What you're referring to is probably when writing the List like this:
```
dstintf:
 - name: VLAN10
   name: VLAN20
```
Which would then be overridden as expected. When using this format, the rules return a 'changed' but only the last item is set.

The error posted on the initial description appears when writing actual lists like this:
```
dstintf:
  - name: VLAN10
  - name: VLAN20
```
Ansible will fail with the error described above ("msg": "Error in repo").

**Comment by adgerth** [comment]
@JieX19 
Hate to bother you again but any news?

**Comment by JieX19** [comment]
Hi @adgerth,

I'm sorry for the late reply. Can you please help check the VDOM mode? Is your device on split task vdom mode? It seems that you don't have access to root. 

Also, please send me a copy of the job log and debugging log for the executable playbook you used (only one value for each attribute).

Thanks!

**Comment by adgerth** [comment]
Hi @JieX19 

no problem. I've had to check back with the team responsible for the firewalls. The Device is in multi-vdom mode and the user I'm using does not have any access to the root vdom. This is by design by the network team due to separation.

I've attached the current ansible log (job.log) and the debug logs from the fortinet (debug.log) when using single items instead of lists in 'srcintf', 'dstintf', 'srcaddr', and 'dstadr'.
[debug.log](https://github.com/fortinet-ansible-dev/ansible-galaxy-fortios-collection/files/6537596/debug.log)
[job.log](https://github.com/fortinet-ansible-dev/ansible-galaxy-fortios-collection/files/6537597/job.log)
